node/resolvers/search/index.ts

```diff
@@ -122,6 +126,22 @@ export const fieldResolvers = {
   ...productPriceRangeResolvers,
 }
 
+const getCompatibilityArgs = async <T extends QueryArgs>(ctx: Context, args: T) => {
+  const { clients: {vbase, search} } = ctx
+  const compatArgs = isLegacySearchFormat(args)? args: await toCompatibilityArgs(vbase, search, args)
+  return compatArgs? {...args, ...compatArgs}: args
+}
+
+const isLegacySearchFormat = ({query, map}: {query: string, map?: string}) => {
```

Comment:
It would be nice to have a comment here so people can understand what "legacy" means